### DISCUSSION TITLE
option to prevent mydumper from stopping replication?

### DISCUSSION THREAD

**Discussion by matthew-lenz-otr** [mydumper/mydumper#1640] [Ideas]
recently ran into an issue that caused mydumper to segfault (david already has committed a fix).  It didn't clean up after itself when this happened and what I didn't realize is that it had stopped replication on the replica.  Fortunately my monitoring notified me but I'm honestly curious what real benefit stopping replication has when there is already locking in place.

**Comment by davidducos** [comment]
Hi @matthew-lenz-otr ,
We are stopping the replica to start the backup in a consistent point in time, as we don't know if you are going to build a new replica based on the current replica server or from its source server. 
On the other hand, mydumper doesn't know if you are going to backup transactional tables or if you will include non-transactional tables unless you use `--trx-consistency-only` which the `START REPLICA` command is send after the binlog position is written on the metadata file.
So, if you are backing up InnoDB tables only, you can use `--trx-consistency-only` and avoid any issue.
